cpp/include/Ice/ObjectAdapter.h

```diff
@@ -31,18 +31,12 @@ namespace Ice
     public:
         virtual ~ObjectAdapter();
 
-        /**
-         * Get the name of this object adapter.
-         * @return This object adapter's name.
-         * @remark This function remains usable after the object adapter has been deactivated.
-         */
-        [[nodiscard]] virtual std::string getName() const = 0;
+        /// Gets the name of this object adapter.
+        /// @return This object adapter's name.
+        [[nodiscard]] virtual const std::string& getName() const = 0;
```

Comment:
Should we add noexcept here too, like for getCommunicator?

Reply:
Yes, fixed now.